packages/block-library/src/heading/style.scss

```diff
@@ -7,7 +7,8 @@
 	h6 {
 		background-color: var(--wp--color--background);
 		color: var(--wp--color--text);
-		line-height: var(--wp--typography--line-height);
+		line-height: var(--wp--typography--line-height, var(--wp--typography--line-height-heading));
```

Comment:
Note that, the line-height hook that is shown in the block inspector is using `--wp--typography--line-height`, which is added as an inline style. If not present, it'll fall back to the global variable. This may need iteration in a subsequent PR, we may want to consolidate everything to general variables ([potential direction](https://href.li/?https://github.com/WordPress/gutenberg/pull/20290), although needs discussion).